website/content/api-docs/features/consistency.mdx

```diff
@@ -37,15 +37,101 @@ The three read modes are:
+
+</Tab>
+<Tab heading="Consistent reads, all writes">
+
+[![consistent consistency mode read, any write request path](/img/consistency-modes/consistency-mode-consistent.png)](/img/consistency-modes/consistency-mode-consistent.png)
+
+</Tab>
+</Tabs>
+
+### Cross-Datacenter Request Behavior
+
+When making a request across federated Consul datacenters, requests are forwarded from
+a local server to any remote server. Once in the remote datecenter, the request path
+is the same as a [local request with the same consistency mode](#intra-datacenter-request-behavior).
+
+[![consistency mode behavior across Consul datacenters](/img/consistency-modes/consistency-mode-cross-datacenter.png)](/img/consistency-modes/consistency-mode-cross-datacenter.png)
+
+## Usage
+
+### Consistency Mode Defaults and Overriding
+
+#### Consul DNS Queries
+
+When DNS queries are issued to [Consul's DNS interface](/docs/discovery/dns),
+Consul uses the `stale` consistency mode by default when interfacing with the
+underlying Consul service discovery HTTP APIs
+([catalog](/api-docs/catalog), [health](/api-docs/health), and [query](/api-docs/query)).
+To set an upper limit on how stale results are allowed to be,
+specify [`dns_config.max_stale`](/docs/agent/options#max_stale)
+in the configuration of the Consul agent serving DNS queries.
+
+If stronger consistency is required at the expense of increased latency,
+set [`dns_config.allow_stale`](/docs/agent/options#allow_stale) to `false`
+to make Consul use the `default` consistency mode when interfacing with the
+underlying Consul service discovery HTTP APIs.
+
+The consistency mode underlying Consul DNS queries cannot be overridden on a
+per-request basis.
+
+#### Consul HTTP API Queries
+
+All HTTP API read requests use the `default` consistency mode by default
+unless overridden on a per-request basis.
+
+Only the service discovery HTTP API endpoints
+([catalog](/api-docs/catalog), [health](/api-docs/health), and [query](/api-docs/query))
+allow changing the consistency mode used by default. By setting
+[`discovery_max_stale`](/docs/agent/options#discovery_max_stale)
+to a value greater than zero in the configuration of the Consul agent receiving requests,
+the `stale` consistency mode will be used by default unless the data is sufficiently stale.
+Setting a positive value for [`discovery_max_stale`](/docs/agent/options#discovery_max_stale)
+is analogous to changing the default consistency mode for
+[Consul DNS queries](#consul-dns-queries) by setting [`dns_config.max_stale`](/docs/agent/options#max_stale).
+
+The HTTP API documentation for each endpoint lists which consistency modes can be used
+to override the default on a per-request basis. To use a supported consistency mode
+other than `default` for a request,
+include the desired consistency mode—`stale` or `consistent`—as a
```

Comment:
If `discovery_max_stale` is >0, is it accurate to say that the consistency mode is `stale` (assuming the Raft entry log index isn't too far behind the leader)?

And if so, if `discovery_max_stale` is in use, does that mean there's no way to force the `default` consistency mode on a request? If I understand correctly, there's no query param that sets `default` consistency mode.

Reply:
@banks : given your summary, it sounds like the start of the `*max_stale` should be something like **not recommended for most use cases** along with a brief description of why (a condensed version of your first bullet in the summary).

> Caveats: there is a reasonable chance I've misremembered some of these details and the above is technically inaccurate in some way.

Does this caveat apply to the summary? Or the technical details backing the summary? We probably don't need those full details on the docs site, but if the **consequence** of setting those params is accurate, we should definitely document that clearly.

Reply:
> Does this caveat apply to the summary?

Caveat applies to the detailed analysis here since it's be at least a year or more since I actually debugged this issue so could have mis-remembered. I'm reasonably sure the outcome is the same: we should warn against setting these and the biggest reason is the load-amplification one. If we write up the exact behaviour in the examples in some longer form answer though, that probably warrants a little more fact checking of how Consul actually behaves. Even if I'm broadly right, there could be other complications I'm unaware of, especially around timeouts and retries.

Reply:
Here's the caveat I added to `dns_config.allow_stale`. I added a similar caveat for `discovery_max_stale`. If this looks good, I can also apply a caveat on the agent configuration page entries for these params.

![image](https://user-images.githubusercontent.com/85913323/158252966-4d0a96d0-8061-4dc8-8ef4-177efb97138a.png)